superset/connectors/druid/models.py

```diff
@@ -1621,8 +1623,9 @@ def query_datasources_by_name(
     ) -> List["DruidDatasource"]:
         return (
             session.query(cls)
-            .filter_by(cluster_name=database.id)
-            .filter_by(datasource_name=datasource_name)
+            .join(DruidCluster)
+            .filter(cls.datasource_name == datasource_name)
+            .filter(DruidCluster.cluster_name == database.name)
```

Comment:
Why isn't this `DruidCluster.id == database.id`?

Reply:
I am not sure whether database.id has anything to do with cluster.id (it doesn't seem to be related in our database).

Reply:
Grokking through the [code](https://github.com/apache/incubator-superset/search?utf8=%E2%9C%93&q=query_datasources_by_name&type=) it seems like this is only called [here](https://github.com/apache/incubator-superset/blob/e6be51953cad5f2872ea8effa40cd1060158f058/superset/security/manager.py#L322) which is SQL specific. It seems like this method should just return an empty list.

@mistercrunch probably has the most context regarding the connection registry.